portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -22,11 +25,27 @@
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
-public class DataServlet extends HttpServlet {
+public final class DataServlet extends HttpServlet {
+
+  private List<String> comments = new ArrayList<>();
```

Comment:
Use `private final ` here. This signals to the reader that this member variable will never be reassigned.